src/larecs/world.mojo

```diff
@@ -1249,33 +1175,45 @@ struct World[*component_types: ComponentType](Copyable, Movable, Sized):
         old_archetype_mask = old_archetype[].get_mask()
 
         @parameter
-        if remove_some:
-            if not old_archetype_mask.contains(BitMask(runtime_remove_ids[])):
-                raise WorldErrors.missing_components_for_removal_entity
+        if rem_size:
+
+            @parameter
+            if remove_some:
+                if not old_archetype_mask.contains(BitMask(remove_ids[])):
+                    raise Error(
+                        "Entity does not have one of the components to remove."
```

Comment:
This could be misunderstood in terms of that only one is required. Maybe better "Entity is missing one of the components to remove."?